lib/index.js

```diff
@@ -119,10 +119,20 @@ module.exports = class PrometheusMetricsConsumer extends Writable {
                 registry: this.registry,
             });
         }
-        this.registry
-            .getSingleMetric(name)
-            .labels(...labelValues)
-            .observe(time);
+
+        if (metric.type && metric.type === 7) {
```

Comment:
I think you will want to handle summaries and histograms so type 7 here and type 5 in the previous block maybe?

Reply:
Good point